maven-plugin/src/main/java/org/openmrs/maven/plugins/Run.java

```diff
@@ -73,13 +81,113 @@ public void executeTask() throws MojoExecutionException, MojoFailureException {
 			}
 		}
 
+		try {
+			File tempFolder = createTempReactorProject(server);
```

Comment:
I would start from putting an if(project.hasWatchedModules()) to make it clear for someone reading the code to ignore the whole try block if the list is empty.